### PR TITLE
Upgrade: patch rancher shell-image to v0.1.26

### PR DESCRIPTION
**Pull request by w13915984028**
**IMPORTANT: Please do not create a Pull Request without creating an issue first.**

**Problem:**

https://github.com/harvester/harvester/issues/6283

The rancher shell default image v0.1.24 has CVE issues.

**Solution:**

Patch Rancher shell version to v0.1.26

**Related Issue:**
https://github.com/harvester/harvester/issues/6283

Related installer PR: https://github.com/harvester/harvester-installer/pull/813

**Test plan:**

```
1. not set, patch
Rancher shell-image is  , will be patched to v0.1.26
setting.management.cattle.io/shell-image patched
NAME          VALUE
shell-image   rancher/shell:v0.1.26


2. lower version, patch
Rancher shell-image is rancher/shell:v0.1.24, will be patched to v0.1.26
setting.management.cattle.io/shell-image patched
NAME          VALUE
shell-image   rancher/shell:v0.1.26

3. updated, no patch
Rancher shell-image is rancher/shell:v0.1.26, patch is not needed
```

### CODE REVIEW

File: package/upgrade/upgrade_manifests.sh [hunk 1]
```diff
@@ -245,6 +245,16 @@ upgrade_rancher() {
     exit 1
   fi
 
+  # When upgrading to Harvester v1.3.2, patch Rancher shell-image to v0.1.26 if necessary
+  local shellimage=$(kubectl get settings.management.cattle.io shell-image -ojsonpath='{.value}')
+  if [[ "$shellimage" = "" ]] || [[ "$shellimage" < "rancher/shell:v0.1.26" ]]; then
```
Comment (starbops) [inline package/upgrade/upgrade_manifests.sh:250]: Is it possible to meet the following scenario?

```
shellimage=rancher/shell:v0.1.3
if [[ "$shellimage" = "" ]] || [[ "$shellimage" < "rancher/shell:v0.1.26" ]]; then 
  echo need patch
fi
# will not patch
```

I'm not sure if we need to handle all the cases. If this is just a quick fix, assuming the before-upgrade will always use `rancher/shell:v0.1.24`, then we're good.

Comment (w13915984028) [reply package/upgrade/upgrade_manifests.sh:250]: This is a specific patch only on Harvester v1.3.2, suppose it is upgraded from v1.3.0/v1.3.1/v1.3.2, the shell image is  like `v0.1.22`, `v0.1.24`. 

A far previous version like `v0.1.14` was release on 2021. https://github.com/rancher/shell/releases/tag/v0.1.14, no need to worry about `v0.1.3`; and there is also no version like `v0.1.31`, the `v0.1.26` is the currently last version of `v0.1.2*`.

Recently, Rancher release new major version of shell, it starts from v0.2.1.

Comment (bk201) [reply package/upgrade/upgrade_manifests.sh:250]: Similar question to the [new cluster case](https://github.com/harvester/harvester-installer/pull/813#issuecomment-2290897219). If we modify the setting to `v0.1.26` here, when the Rancher is upgraded I guess the setting is still v0.1.26 and will need some clean-up.

**Review by starbops — APPROVED** [review]
Tested, LGTM!

Could we have another issue to track this and harvester/harvester-installer#813 for future patches/improvements?

File: package/upgrade/upgrade_manifests.sh (hunk 1 above)
Comment (w13915984028) [reply package/upgrade/upgrade_manifests.sh:250]: Log a new issue https://github.com/harvester/harvester/issues/6352 to drop this potential patch on v1.3.2 cluster.

**Review by bk201 — APPROVED** [review]

**Comment by w13915984028** [comment]
This patch will be dropped in Harvester v1.4.0

PR: https://github.com/harvester/harvester/pull/6353
ISSUE:  https://github.com/harvester/harvester/issues/6352

**Pull request merged**